CTests/Descriptor/test_bond_env.cpp

```diff
@@ -0,0 +1,202 @@
+#include "gtest/gtest.h"
+#include "Descriptor/single_bond.h"
+#include "Environment/local_environment.h"
+#include "Structure/structure.h"
+#include "Descriptor/cutoffs.h"
+#include "Descriptor/radial.h"
+#include "Descriptor/descriptor.h"
+#include <iostream>
+#include <cmath>
+#include <Eigen/Dense>
+
+class BondEnv : public ::testing::Test{
+    protected:
+
+    double delta = 1e-8;
+    int noa = 5;
+    int nos = 5;
+
+    Eigen::MatrixXd cell {3, 3}, cell_2 {3, 3};
+    std::vector<int> species {0, 2, 1, 3, 4};
+    Eigen::MatrixXd positions_1 {noa, 3}, positions_2 {noa, 3};
+
+    Structure struc1, struc2;
+    LocalEnvironment env1, env2;
+
+    double rcut = 3;
+    std::vector<double> many_body_cutoffs {rcut};
+ 
+    // Prepare cutoff.
+    std::vector<double> cutoff_hyps;
+    void (*cutoff_function)(double *, double, double, std::vector<double>) =
```

Comment:
@dmclark17 is leaving, you can use `std::function`s now

Reply:
You should use templates to avoid any run-time redirection